embassy-rp/src/watchdog.rs

```diff
@@ -0,0 +1,111 @@
+//! Watchdog
+//!
+//! The watchdog is a countdown timer that can restart parts of the chip if it reaches zero. This can be used to restart the
+//! processor if software gets stuck in an infinite loop. The programmer must periodically write a value to the watchdog to
+//! stop it from reaching zero.
+//!
+//! Credit: based on `rp-hal` implementation (also licensed Apache+MIT)
+
+use core::marker::PhantomData;
+
+use embassy_time::Duration;
+
+use crate::pac;
+use crate::peripherals::WATCHDOG;
+
+/// Watchdog peripheral
+pub struct Watchdog<'d> {
+    phantom: PhantomData<&'d WATCHDOG>,
+    load_value: u32, // decremented by 2 per tick (µs)
+}
+
+impl<'d> Watchdog<'d> {
+    /// Create a new `Watchdog`
+    pub fn new(_watchdog: WATCHDOG) -> Self {
+        Self {
+            phantom: PhantomData,
+            load_value: 0,
+        }
+    }
+
+    /// Start tick generation on clk_tick which is driven from clk_ref.
+    ///
+    /// # Arguments
+    ///
+    /// * `cycles` - Total number of tick cycles before the next tick is generated.
+    ///   It is expected to be the frequency in MHz of clk_ref.
+    pub fn enable_tick_generation(&mut self, cycles: u8) {
+        const WATCHDOG_TICK_ENABLE_BITS: u32 = 0x200;
+        unsafe {
+            let watchdog = pac::WATCHDOG;
+            watchdog
+                .tick()
+                .write_value(pac::watchdog::regs::Tick(WATCHDOG_TICK_ENABLE_BITS | cycles as u32))
```

Comment:
could you use the fields from the PAC? `tick().write(|w| { w.set_enable(true); w.set_ccycles(cycles));`